### ISSUE TITLE
ThingsBoard upgrade from 2.1.0 to 2.2.0 is failed!

### ISSUE THREAD

**Issue by smart247** [thingsboard/thingsboard#1323]
sudo /usr/share/thingsboard/bin/install/upgrade.sh --fromVersion=2.0.0
20:19:08,654 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback-test.xml]
20:19:08,660 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback.groovy]
20:19:08,660 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Found resource [logback.xml] at [file:/usr/share/thingsboard/conf/logback.xml]
20:19:08,708 |-WARN in ch.qos.logback.classic.LoggerContext[default] - Resource [logback.xml] occurs multiple times on the classpath.
20:19:08,709 |-WARN in ch.qos.logback.classic.LoggerContext[default] - Resource [logback.xml] occurs at [jar:file:/usr/share/thingsboard/bin/thingsboard.jar!/BOOT-INF/lib/queue-2.2.0.jar!/logback.xml]
20:19:08,709 |-WARN in ch.qos.logback.classic.LoggerContext[default] - Resource [logback.xml] occurs at [file:/usr/share/thingsboard/conf/logback.xml]
20:19:09,885 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - debug attribute not set
20:19:09,894 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.rolling.RollingFileAppender]
20:19:10,053 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [fileLogAppender]
20:19:10,224 |-INFO in c.q.l.core.rolling.SizeAndTimeBasedRollingPolicy@1329552164 - setting totalSizeCap to 3 GB
20:19:10,271 |-INFO in c.q.l.core.rolling.SizeAndTimeBasedRollingPolicy@1329552164 - Archive files will be limited to [100 MB] each.
20:19:10,480 |-INFO in c.q.l.core.rolling.SizeAndTimeBasedRollingPolicy@1329552164 - No compression will be used
20:19:10,488 |-INFO in c.q.l.core.rolling.SizeAndTimeBasedRollingPolicy@1329552164 - Will use the pattern /var/log/thingsboard/thingsboard.%d{yyyy-MM-dd}.%i.log for the active file
20:19:10,512 |-INFO in ch.qos.logback.core.rolling.SizeAndTimeBasedFNATP@15aeb7ab - The date pattern is 'yyyy-MM-dd' from file name pattern '/var/log/thingsboard/thingsboard.%d{yyyy-MM-dd}.%i.log'.
20:19:10,512 |-INFO in ch.qos.logback.core.rolling.SizeAndTimeBasedFNATP@15aeb7ab - Roll-over at midnight.
20:19:10,538 |-INFO in ch.qos.logback.core.rolling.SizeAndTimeBasedFNATP@15aeb7ab - Setting initial period to Tue Dec 11 19:33:10 UTC 2018
20:19:10,577 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
20:19:10,840 |-INFO in ch.qos.logback.core.rolling.RollingFileAppender[fileLogAppender] - Active log file name: /var/log/thingsboard/thingsboard.log
20:19:10,840 |-INFO in ch.qos.logback.core.rolling.RollingFileAppender[fileLogAppender] - File property is set to [/var/log/thingsboard/thingsboard.log]
20:19:10,851 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.thingsboard.server] to INFO
20:19:10,856 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [akka] to INFO
20:19:10,856 |-INFO in ch.qos.logback.classic.joran.action.RootLoggerAction - Setting level of ROOT logger to INFO
20:19:10,857 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [fileLogAppender] to Logger[ROOT]
20:19:10,859 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - End of configuration.
20:19:10,862 |-INFO in ch.qos.logback.classic.joran.JoranConfigurator@7b23ec81 - Registering current configuration as safe fallback point

 ===================================================
 :: ThingsBoard ::       (v2.2.0)
 ===================================================

log4j:WARN No appenders could be found for logger (hsqldb.db.HSQLDB679EED3791.ENGINE).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
Starting ThingsBoard Upgrade from version 2.0.0 ...
Upgrading ThingsBoard from version 2.0.0 to 2.1.1 ...
Updating schema ...
Schema updated.
Upgrading ThingsBoard from version 2.1.1 to 2.1.2 ...
Dumping entity views ...
Entity views dumped.
Updating schema ...
Schema updated.
Restoring entity views ...
Entity views restored.
Upgrading ThingsBoard from version 2.1.3 to 2.2.0 ...
Updating schema ...
Unexpected error during ThingsBoard installation!
java.sql.SQLSyntaxErrorException: user lacks privilege or object not found: PUBLIC.COMPONENT_DESCRIPTOR
        at org.hsqldb.jdbc.JDBCUtil.sqlException(Unknown Source)
        at org.hsqldb.jdbc.JDBCUtil.sqlException(Unknown Source)
        at org.hsqldb.jdbc.JDBCStatement.fetchResult(Unknown Source)
        at org.hsqldb.jdbc.JDBCStatement.execute(Unknown Source)
        at org.thingsboard.server.service.install.SqlDatabaseUpgradeService.loadSql(SqlDatabaseUpgradeService.java:167)
        at org.thingsboard.server.service.install.SqlDatabaseUpgradeService.upgradeDatabase(SqlDatabaseUpgradeService.java:156)
        at org.thingsboard.server.install.ThingsboardInstallService.performInstall(ThingsboardInstallService.java:107)
        at org.thingsboard.server.ThingsboardInstallApplication.main(ThingsboardInstallApplication.java:43)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48)
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:87)
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:50)
        at org.springframework.boot.loader.PropertiesLauncher.main(PropertiesLauncher.java:521)
Caused by: org.hsqldb.HsqlException: user lacks privilege or object not found: PUBLIC.COMPONENT_DESCRIPTOR
        at org.hsqldb.error.Error.error(Unknown Source)
        at org.hsqldb.error.Error.error(Unknown Source)
        at org.hsqldb.SchemaManager.getUserTable(Unknown Source)
        at org.hsqldb.ParserDDL.compileAlterTable(Unknown Source)
        at org.hsqldb.ParserDDL.compileAlter(Unknown Source)
        at org.hsqldb.ParserCommand.compilePart(Unknown Source)
        at org.hsqldb.ParserCommand.compileStatements(Unknown Source)
        at org.hsqldb.Session.executeDirectStatement(Unknown Source)
        at org.hsqldb.Session.execute(Unknown Source)
        ... 14 common frames omitted
Unexpected error during ThingsBoard installation!
ThingsBoard upgrade failed!

How can I fix it?

**Comment by dedors** [comment]
Same problem (cant upgrade from 2.1.3 to 2.2.0), different error:

 ===================================================
 :: ThingsBoard ::       (v2.2.0)
 ===================================================

Starting ThingsBoard Upgrade from version 2.1.3 ...
Upgrading ThingsBoard from version 2.1.3 to 2.2.0 ...
Updating system data...
Unexpected error during ThingsBoard installation!
java.lang.NullPointerException: null
        at org.thingsboard.server.dao.nosql.CassandraAbstractDao.getSession(CassandraAbstractDao.java:64)
        at org.thingsboard.server.dao.nosql.CassandraAbstractDao.executeAsync(CassandraAbstractDao.java:118)
        at org.thingsboard.server.dao.nosql.CassandraAbstractDao.execute(CassandraAbstractDao.java:108)
        at org.thingsboard.server.dao.nosql.CassandraAbstractDao.executeRead(CassandraAbstractDao.java:89)
        at org.thingsboard.server.dao.nosql.CassandraAbstractModelDao.findOneByStatement(CassandraAbstractModelDao.java:98)
        at org.thingsboard.server.dao.widget.CassandraWidgetsBundleDao.findWidgetsBundleByTenantIdAndAlias(CassandraWidgetsBundleDao.java:67)
        at org.thingsboard.server.dao.widget.WidgetsBundleServiceImpl.findWidgetsBundleByTenantIdAndAlias(WidgetsBundleServiceImpl.java:87)
        at org.thingsboard.server.service.install.DefaultSystemDataLoaderService.deleteSystemWidgetBundle(DefaultSystemDataLoaderService.java:159)
        at org.thingsboard.server.install.ThingsboardInstallService.performInstall(ThingsboardInstallService.java:111)
        at org.thingsboard.server.ThingsboardInstallApplication.main(ThingsboardInstallApplication.java:43)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48)
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:87)
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:50)
        at org.springframework.boot.loader.PropertiesLauncher.main(PropertiesLauncher.java:521)
Unexpected error during ThingsBoard installation!

**Comment by vparomskiy** [comment]
Please execute upgrade again and share install.log file

**Comment by dedors** [comment]
I removed my previous post here after discovering that the error was on my side. It turned out, that with the previous docker installation (via docker-compose) I was pretty sure I set it up to use cassandra, while it only used a local file (HSQLDB).
Transition to Thingsboard 2.2 ment that instead of the docker-compose it now has a single container, so I had to move the database directory to the new container data path. The old one contained 3 folders for all 3 database types. Of course assuming im running cassandra, I copied the wrong data. What was confusing that the never-used database had a size 100mb, concluding that this is the active db I used before.

On the way to figure that out I tried to migrate on a direct Raspi installation. I thing the error message was the same as OP, I'm not saying its the same problem but worth checking it out.

I hope thingsboard will have a easy migration path from HSQLDB to cassandra one day.
